core/app/models/spree/order_inventory.rb

```diff
@@ -48,12 +48,18 @@ def remove(quantity, shipment = nil)
     # Returns either one of the shipment:
     #
     # first unshipped that already includes this variant
-    # first unshipped that's leaving from a stock_location that stocks this variant
-    def determine_target_shipment
+    # first unshipped that's leaving from a stock_location that stocks this variant, with availability check
+    # first unshipped that's leaving from a stock_location that stocks this variant, without availability check
+    def determine_target_shipment(quantity)
```

Comment:
I understand what you mean. But in this context, I don't know what could be a good default value.

This method is called from here:
```ruby
        if desired_quantity > 0
          shipment ||= determine_target_shipment(desired_quantity)
# ...
    def determine_target_shipment(quantity)
# ...
```

So, the number`0` would do a meaningless default. Would `1` be more meaningful? Yes, but again not completely... I don't feel any default number would really make sense, unfortunately.

What do you think?

Reply:
My 2 cents: we shouldn't be too concerned about changing the signature of a private method IMO, since those are expected to change without notice. Of course we should avoid it if possible, but it doesn't seem like there's a good default here.

Reply:
Yes, you are probably both right. We should add something on the CHANGELOG about this, adding the `Important Change` label to this PR so that we'll recall this when releasing the next version.

Reply:
I'm fine changing it because it's `private` too. Those methods shouldn't be used by anyone other than the internal workings of the public methods. Adding to the `CHANGELOG` is smart just in case anyone has overwritten that method for whatever reason.

👍